python/ray/util/ray_executor.py

```diff
@@ -0,0 +1,237 @@
+                    return fn()
+
+            actors = [
+                ExecutorActor.options(name=f"actor-{i}").remote()
+                for i in range(max_workers)
+            ]
+            self.__actor_pool = ray.util.ActorPool(actors)
+
+        self.context = ray.init(ignore_reinit_error=True, **kwargs)
+
+    def submit(self, fn, *args, **kwargs):
+        """Submits a callable to be executed with the given arguments.
+
+        Schedules the callable to be executed as `fn(*args, **kwargs)` and returns
+        a Future instance representing the execution of the callable.
+
+        Returns:
+            A Future representing the given call.
+
+        Usage example:
+
+        .. code-block:: python
+
+            with RayExecutor() as ex:
+                future_0 = ex.submit(lambda x: x * x, 100)
+                future_1 = ex.submit(lambda x: x + x, 100)
+                result_0 = future_0.result()
+                result_1 = future_1.result()
+        """
+        self._check_shutdown_lock()
+        fn_curried = partial(fn, *args, **kwargs)
+        if self.__actor_pool:
+            self.__actor_pool.submit(
+                lambda a, _: a.actor_function.remote(fn_curried), None
+            )
+            oref = self.__actor_pool._index_to_future[
+                self.__actor_pool._next_task_index - 1
+            ]
+        else:
+            oref = self.__remote_fn.options(name=fn.__name__).remote(fn_curried)
+        future = oref.future()
+        self._futures.append(future)
+        return future
+
+    def map(self, fn, *iterables, timeout=None, chunksize=1):
+        """Returns an iterator equivalent to `map(fn, iter)`.
+
+        Args:
+            fn: A callable that will take as many arguments as there are
+                passed iterables.
+            timeout: The maximum number of seconds to wait. If None, then there
+                is no limit on the wait time.
+            chunksize: The size of the chunks the iterable will be broken into
+                before being passed to a child process.
+
+        Returns:
+            An iterator equivalent to: `map(func, *iterables)` but the calls may
+            be evaluated out-of-order.
+
+        Raises:
+            TimeoutError: If the entire result iterator could not be generated
+                before the given timeout.
+            Exception: If `fn(*args)` raises for any values.
+
+        Usage example:
+
+        .. code-block:: python
+
+            with RayExecutor() as ex:
+                futures = ex.map(lambda x: x * x, [100, 100, 100])
+                results = [future.result() for future in futures()]
+
+        """
+        self._check_shutdown_lock()
+        results_list = []
+        for chunk in self._get_chunks(*iterables, chunksize=chunksize):
+            if self.__actor_pool:
+                results = self.__actor_pool.map(
+                    lambda a, v: a.actor_function.remote(partial(fn, *v)), chunk
```

Comment:
```suggestion
                    lambda a, v: a.options(name=<name of fn>).actor_function.remote(partial(fn, *v)), chunk
```
Not sure if this API exists, but can you check it?

Reply:
Looks like that's an `ActorHandle` referring to an existing Actor.